### PR TITLE
DAOS-3119 test: Standardizing on server/agent yaml file specifications

### PR DESCRIPTION
**Pull request by Puspalogan**
- Modified all functional test yaml files to have standardized
   test_servers and test_clients configuration and removed the
   test_machines tag.
 - Updated few tests to use apricot's TestWithServers
 - Updated python scripts to use test_servers where necessary as
   opposed to test_machines as it was removed from yaml files
 - Changed the host names from boro-* to either server-* or
   client-* to have generic naming convention.

Signed-off-by: Logan Sundaram <logan.sundaram@intel.com>

### CODE REVIEW

**Comment by Puspalogan** [comment]
> LGTM if we are handling moving `io/romio.py`, `object/object_integrity.py`, `pool/bad_connect.py` and `pool/multi_server_create_delete_test.py` to use `TestWithServers` in a separate PR.

I understand that 'TestWithServers' fix may be beyond the scope of the original story. If there isn't another story for that one we'd have to create one. To confirm, following are the files that needs to be revered.

- io/romio.py
- pool/bad_exclude.py
- pool/multiple_creates_test.py

Other files such as,

- object/object_integrity.py
- pool/bad_connect.py
- pool/multi_server_create_delete_test.py

must still go in this PR to support the yaml files with 'test_servers' tag.

**Comment by phender** [comment]
> > LGTM if we are handling moving `io/romio.py`, `object/object_integrity.py`, `pool/bad_connect.py` and `pool/multi_server_create_delete_test.py` to use `TestWithServers` in a separate PR.
> 
> I understand that 'TestWithServers' fix may be beyond the scope of the original story. If there isn't another story for that one we'd have to create one. To confirm, following are the files that needs to be revered.

My apologies for the confusion, I should have double checked DAOS-3119 before adding my comment.  It turns out that this PR is actually the one that includes moving all the tests to use `TestWithServers`:

"It would also be beneficial for all tests specifying servers and agents to inherit the TestWithServers apricot class to utilize the same common code for reading these values from the yaml files. "

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -63,7 +63,6 @@ def __del__(self):
 TEST_LOG_FILE_YAML = "./data/daos_avocado_test.yaml"
 BASE_LOG_FILE_YAML = "./data/daos_server_baseline.yaml"
 SERVER_KEYS = (
-    "test_machines",
```
Comment (phender) [inline src/tests/ftest/launch.py]: In addition the `daos_servers` entry can be removed - it used to be used by soak.yaml

Comment (Puspalogan) [reply src/tests/ftest/launch.py]: Done

**Comment by sylviachanoiyee** [comment]
The changes look ok but the ior_small test is failing. If i'm following the discussion on ior_small failure correctly, it has nothing to do with this patch?
I'm going to approve this patch and let the gatekeeper's decision as to whether to land it.

**Comment by Puspalogan** [comment]
> The changes look ok but the ior_small test is failing. If i'm following the discussion on ior_small failure correctly, it has nothing to do with this patch?
> I'm going to approve this patch and let the gatekeeper's decision as to whether to land it.

Yes, Thank you !

**Comment by Puspalogan** [comment]
> Please resolve the conflicts before merge.

Done

**Pull request merged**